### ISSUE TITLE
Table autosize issues with percentage width and colspan

### ISSUE THREAD

**Issue by daniel-johnsson** [Aymkdn/html-to-pdfmake#151]
First of all thanks for a great package, this really is a lifesaver.

I found some issues when tinkering around with tables, I already have fixed these issues manually with some pre/post-processing. Just wanted to report them so they might be resolved in a nicer fashion.

If the width is set to a percentage it will be set to auto if colspan is set:

With colspan
```
<table>
  <tr>
    <td style="width: 20%" colspan="1">Cell A1</td>
    <td>Cell A2</td>
  </tr>
</table>

```
`"widths": ["auto", "auto"]`

Without colspan
```
<table>
  <tr>
    <td style="width: 20%">Cell A1</td>
    <td>Cell A2</td>
  </tr>
</table>

```
`"widths": ["20%", "auto"]`

I assume that it breaks on [this line](https://github.com/Aymkdn/html-to-pdfmake/blob/master/index.js#L273) and is converted to `auto`

Second issue is that `auto` is only converted too `*` if all widths are `auto`. I would assume that we want to convert all `auto` to a format that works in pdfMake. 

My suggestion would be to [change the code here](https://github.com/Aymkdn/html-to-pdfmake/blob/master/index.js#L306)
Replaced with: 
``` javascript
if (fullWidth) tableWidths=tableWidths.map(function() { return w==='auto' ? '*' : w; });
```

Let me know if I can assist in any way :)

**Comment by Aymkdn** [comment]
> If the width is set to a percentage it will be set to auto if colspan is set

Actually, only the HTML attribute `width` is really supported when there is a `colspan`. Dealing with the width+colspan is difficult… So I won't change the support until someone really asks for a specific example.

In the meantime, for your case, I changed the code to ignore `colspan < 2` (not available in current version as it's a very minor change and it will only be released with the next one).

> I would assume that we want to convert all `auto` to a format that works in pdfMake.

I'm not sure to understand what you mean? `auto` is a valid value for PDFMake:
![image](https://user-images.githubusercontent.com/946315/194513761-1b218746-a4af-401d-b59a-c60ad76b7c1c.png)

**Comment by Aymkdn** [comment]
Ah OK I understand now. I've just published the v2.4.7 with the changes.

Thanks
